src/config/Config.h

```diff
@@ -0,0 +1,195 @@
+#ifndef CASBIN_CPP_CONFIG_CONFIG
+#define CASBIN_CPP_CONFIG_CONFIG
+
+#include <string>
+#include <sstream>
+#include <unordered_map>
+#include <mutex>
+#include <fstream>
+#include <stdio.h>
+#include <algorithm>
+
+#include "./ConfigInterface.h"
+#include "../exception/IOException.h"
+#include "../exception/IllegalArgumentException.h"
+#include "../util/trim.h"
+#include "../util/ends_with.h"
+#include "../util/split.h"
+
+#endif
+
+using namespace std;
+
+mutex mtx_lock;
+
+class Config : public ConfigInterface {
+    private: 
+        static const string DEFAULT_SECTION;
+        static const string DEFAULT_COMMENT;
+        static const string DEFAULT_COMMENT_SEM;
+        unordered_map < string, unordered_map <string, string> > data;
+
+        /**
+         * addConfig adds a new section->key:value to the configuration.
+         */
+        bool addConfig(string section, string option, string value) {
+            if (!section.compare("")) {
+                section = DEFAULT_SECTION;
+            }
```

Comment:
You have declared and assigned **uninitallized constants** here and at several other places.

Reply:
I have initialised the static constants at the end of the file, please check it out, then to thanks for your efforts

Reply:
It will throw you an error if you initialize them in the class declaration.

Reply:
Sorry brother, i wrote it as i saw that their was no Config.cpp file.
It will be great if you seperate the class declaration and class definition in their seperate files.